library/ssl_tls.c

```diff
@@ -58,6 +58,54 @@
 
 #define PROPER_HS_FRAGMENT 0x75555555
 
+#if defined(MBEDTLS_SSL_TRANSFORM_OPTIMIZE_CIPHERS)
+static int mbedtls_ssl_switch_key( mbedtls_ssl_transform *transform,
+                                   const mbedtls_operation_t operation )
+{
+    unsigned char * key;
+    int ret;
```

Comment:
@AndrzejKurek , lets make it more FI protected(we are checking key integrity inside this function). 
1. Initialize error code to error value
2. Do not return success as default case
3. May be additional error code for line 90 ?